src/main/java/com/google/api/generator/gapic/composer/utils/CommentFormatter.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer.utils;
+
+import com.google.api.generator.engine.ast.JavaDocComment;
+import com.google.common.base.Strings;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CommentFormatter {
+
+  private CommentFormatter() {}
```

Comment:
This empty constructor is not needed.

Reply:
I think this is from sonar recommendation to use private constructor to hide the implicit public one.
https://rules.sonarsource.com/java/RSPEC-1118

Reply:
I see, this is one of the controversial rules though, see this [stackoverflow](https://stackoverflow.com/questions/14398747/how-to-disable-warning-on-sonar-hide-utility-class-constructor) discussion. IMO, it does not provide much value and add unnecessary code, maybe we should turn off this rule? WDYT @meltsufin ?

Reply:
I kind of agree with Sonar here. A private constructor is helpful to avoid mistakes around instantiating the class when it's intended to be used as a collection of static utility methods.

Reply:
I think it makes sense if this is a util class in `gax-java` or a backend of a web app, but maybe not for `gapic-generator-java`, because pretty much everything in the generator is called through `public static` methods in parsers and composers, the whole generator is basically a util. 

There may be unnecessary garbage collection if we accidentally instantiated a util class, which could cause some harm in a server if the util method is call multiple times, but there is no harm for the generator because the generator is run once per jvm.

Reply:
Mike and I discussed offline and we are going to remove this rule from gapic-generator-java, but it's not critically enough to block this PR.